### PR TITLE
PRO-1393 Trading Activity Screen

### PR DESCRIPTION
**Pull request by lalitkumarrock**
https://linear.app/pillarproject/issue/PRO-1393/trading-activity-screen

## Description

- Trading History screen to see token trading history.

## How Has This Been Tested?

- Tested on Android device
- Tested on Simulator 14 Pro

## Screenshots (if appropriate):

https://github.com/pillarwallet/pillarwallet/assets/82652040/c85a25b2-4b09-4ed4-b619-9fd5a32ab292

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/utils/rates.js [hunk 1]
```diff
@@ -107,10 +109,46 @@ export const fiatInvestmentBalance = (balance: BigNumber, rates: RatesPerChain,
...
-export const fiatTokenValue = (tokenValue: number, rates: RatesPerChain, currency: Currency, decimals?: ?number) => {
+export const fiatTokenValue = (tokenValue: number, rates: RatesPerChain, currency: Currency) => {
+  if (isNaN(tokenValue)) return '';
   const nativeAssetRate = rates[nativeAssetPerChain[CHAIN.ETHEREUM].address];
+  const currencySymbol = getCurrencySymbol(currency);
 
   const fiatAmount = (tokenValue * nativeAssetRate?.[currency]) / nativeAssetRate?.[USD];
 
-  return fiatAmount?.toFixed(decimals || 2);
+  const decimalsFiatValue = fiatAmount?.toFixed(fiatAmount > 1 ? 2 : 4);
```
Comment (IAmKio) [inline src/utils/rates.js:119]: Just a question, in what scenario do we need `toFixed` precision to be `4`?

**Review by IAmKio — COMMENTED** [review]
Approved,

File: src/utils/rates.js (hunk 1 above)
Comment (lalitkumarrock) [reply src/utils/rates.js:119]: if fiat value is less than 1. like 0.0553 
if more than 1. like 1.56

Comment (IAmKio) [reply src/utils/rates.js:119]: Got it - cheers

**Review by IAmKio — APPROVED** [review]
LGTM

**Pull request merged**